### ISSUE TITLE
App crash when I want to save a file

### ISSUE THREAD

**Issue by deacon78** [kineapps/flutter_file_dialog#1]
Hello
I tried to use the following method 
`final params = SaveFileDialogParams(sourceFilePath: "path_of_file_to_save");
final filePath = await FlutterFileDialog.saveFile(params: params);
print(filePath);`

But it causes my flutter app to crash.
It opens the file dialog plugin, and it lost connexion with device...

**Comment by kinex** [comment]
Did you try it in Android or in iOS? What is the log output when it crashes?

**Comment by cormalenv** [comment]
Hi @kinex,

I assume `SaveFileDialogParams` is essentially "copying"  as it requires you to have a file in existence in order to provide an valid file path for the method to **_re-save_** it elsewhere. Conceptually then to implement true saving in one's app you would create a `_file` in say `Directory.systemTemp` and then pass that `_file.path` as the `sourceFilePath`. 

Thanks for the work.

**Comment by deacon78** [comment]
Hello

I was trying it on an Android Emulator through Android Studio.

First, I encode a file in zip format, and then I tried to save it where the user wants to (Google Drive or Downloads for example).

Here is my code :
` try {
      debugPrint('Begin to ZIP file');
      var encoder = ZipFileEncoder();
      encoder.create(donneesVoiture.path);
    //Adding files to archive donneesVoiture
      encoder.addFile(fileTableVoiture);
      encoder.addFile(fileTableEntretien);
      encoder.addFile(fileTableHistoriqueEntretiens);
      encoder.addFile(fileTableControleTechnique);
      encoder.addFile(fileTableModeOPEntretien);
      encoder.close();
      debugPrint('zip file created');
      debugPrint('FlutterSaveDialog : to choose where to store the zip file "donneesVoiture" via a file explorer');
      final params = SaveFileDialogParams(sourceFilePath: donneesVoiture.path);
      final filePath = await FlutterFileDialog.saveFile(params: params);
      debugPrint(filePath);
      print(filePath);

    }catch (e) {
      debugPrint('Error while creating ZIP file');
      print(e);
    }`
And I got this error : 
`/FlutterFileDialogPlugin(14505): onMethodCall, method=saveFile
D/FileDialog(14505): saveFile - IN, sourceFileName=/storage/emulated/0/Android/data/com.example.carnet_entretien_virtuel/files/DL-749-YE.zip, mimeTypesFilter=null, localOnly=false
D/FileDialog(14505): saveFile - OUT
E/Surface (14505): getSlotFromBufferLocked: unknown buffer: 0xa1535680
D/EGL_emulation(14505): eglMakeCurrent: 0xab30c220: ver 3 1 (tinfo 0x9f5cb640)
E/Surface (14505): getSlotFromBufferLocked: unknown buffer: 0xb40fcd00
D/FlutterView(14505): Detaching from a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@a1df5b2
D/AndroidRuntime(14505): Shutting down VM
E/AndroidRuntime(14505): FATAL EXCEPTION: main
E/AndroidRuntime(14505): Process: com.example.carnet_entretien_virtuel, PID: 14505
E/AndroidRuntime(14505): java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=19112, result=-1, data=Intent { dat=content://com.android.providers.downloads.documents/document/15 flg=0x43 }} to activity {com.example.carnet_entretien_virtuel/com.example.carnet_entretien_virtuel.MainActivity}: kotlin.KotlinNullPointerException
E/AndroidRuntime(14505): 	at android.app.ActivityThread.deliverResults(ActivityThread.java:3699)
E/AndroidRuntime(14505): 	at android.app.ActivityThread.handleSendResult(ActivityThread.java:3742)
E/AndroidRuntime(14505): 	at android.app.ActivityThread.-wrap16(ActivityThread.java)
E/AndroidRuntime(14505): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1393)
E/AndroidRuntime(14505): 	at android.os.Handler.dispatchMessage(Handler.java:102)
E/AndroidRuntime(14505): 	at android.os.Looper.loop(Looper.java:148)
E/AndroidRuntime(14505): 	at android.app.ActivityThread.main(ActivityThread.java:5417)
E/AndroidRuntime(14505): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(14505): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
E/AndroidRuntime(14505): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
E/AndroidRuntime(14505): Caused by: kotlin.KotlinNullPointerException
E/AndroidRuntime(14505): 	at com.kineapps.flutter_file_dialog.FileDialog.onActivityResult(FileDialog.kt:134)
E/AndroidRuntime(14505): 	at io.flutter.embedding.engine.FlutterEnginePluginRegistry$FlutterEngineActivityPluginBinding.onActivityResult(FlutterEnginePluginRegistry.java:634)
E/AndroidRuntime(14505): 	at io.flutter.embedding.engine.FlutterEnginePluginRegistry.onActivityResult(FlutterEnginePluginRegistry.java:367)
E/AndroidRuntime(14505): 	at io.flutter.embedding.android.FlutterActivityAndFragmentDelegate.onActivityResult(FlutterActivityAndFragmentDelegate.java:546)
E/AndroidRuntime(14505): 	at io.flutter.embedding.android.FlutterActivity.onActivityResult(FlutterActivity.java:594)
E/AndroidRuntime(14505): 	at android.app.Activity.dispatchActivityResult(Activity.java:6428)
E/AndroidRuntime(14505): 	at android.app.ActivityThread.deliverResults(ActivityThread.java:3695)
E/AndroidRuntime(14505): 	... 9 more
I/Process (14505): Sending signal. PID: 14505 SIG: 9
Lost connection to device.`

**Comment by deacon78** [comment]
Same result on a real device

**Comment by kinex** [comment]
Could you please try this with the provided example app. Clone current master and execute `flutter run` in the example directory. Then pick an image file and try to save it to Google Drive. Does this crash also? 

Have you already upgraded your Android project to V2 embedding? See https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects . Not sure if the crash could be related to v1 embedding (although it *should* work also with v1). Example app has been upgraded to v2 embedding.

**Comment by kinex** [comment]
@cormalenv Yes you have understood it correctly. The API name `saveFile` may be slightly confusing, maybe it should be renamed e.g. to `exportFile`. So with this api you can export a file from your app to an user selectable location which user can access (which can be e.g. Downloads folder, Googe Drive etc.). The file to export could be an already existing file in your app data like an image file or you could also create a new file to export in a temporary directory (like a backup zip file or exported data in a CSV file).

**Comment by cormalenv** [comment]
Similar result as @deacon78, Example code below.  v2 embedding in place

```
FATAL EXCEPTION: main
Process: com..., PID: 5389
java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=19112, result=-1, data=Intent { dat=content://com.android.providers.downloads.documents/document/313 flg=0x43 }} to activity {com.../com....MainActivity}: kotlin.KotlinNullPointerException
	at android.app.ActivityThread.deliverResults(ActivityThread.java:4339)
	at android.app.ActivityThread.handleSendResult(ActivityThread.java:4383)
	at android.app.ActivityThread.-wrap19(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1690)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:171)
	at android.app.ActivityThread.main(ActivityThread.java:6627)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:518)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:873)
Caused by: kotlin.KotlinNullPointerException
	at com.kineapps.flutter_file_dialog.FileDialog.onActivityResult(FileDialog.kt:134)
	at io.flutter.embedding.engine.FlutterEnginePluginRegistry$FlutterEngineActivityPluginBinding.onActivityResult(FlutterEnginePluginRegistry.java:634)
	at io.flutter.embedding.engine.FlutterEnginePluginRegistry.onActivityResult(FlutterEnginePluginRegistry.java:367)
	at io.flutter.embedding.android.FlutterActivityAndFragmentDelegate.onActivityResult(FlutterActivityAndFragmentDelegate.java:546)
	at io.flutter.embedding.android.FlutterActivity.onActivityResult(FlutterActivity.java:594)
	at android.app.Activity.dispatchActivityResult(Activity.java:7343)
	at android.app.ActivityThread.deliverResults(ActivityThread.java:4335)
	... 9 more
```

```
import 'package:flutter/material.dart';
import 'dart:io';
import 'package:flutter_file_dialog/flutter_file_dialog.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: SafeArea(
        child: Scaffold(
          body: RaisedButton(
            child: Text('Run'),
            onPressed: () => run(),
          ),
        ),
      ),
    );
  }

  Future<void> run () async {
    File _file = File (Directory.systemTemp.path +  "/test.txt");
    _file.writeAsStringSync('my string contents');
    final params = SaveFileDialogParams(sourceFilePath: _file.path);
    final filePath = await FlutterFileDialog.saveFile(params: params);
  }
}
```

**Comment by kinex** [comment]
@cormalenv Thanks for the details. It seems this issue is related to https://github.com/flutter/flutter/issues/48202

**Comment by kinex** [comment]
Issue fixed in plugin version 0.0.2

**Comment by cormalenv** [comment]
Thanks @kinex
